src/types/redis_bloom_chain.cc

```diff
@@ -0,0 +1,201 @@
+  batch->PutLogData(log_data.Encode());
+
+  std::string sb_chain_meta_bytes;
+  metadata->Encode(&sb_chain_meta_bytes);
+  batch->Put(metadata_cf_handle_, ns_key, sb_chain_meta_bytes);
+
+  std::string bf_key = getBFKey(ns_key, *metadata, metadata->n_filters - 1);
+  batch->Put(bf_key, block_split_bloom_filter.GetData());
+
+  return storage_->Write(storage_->DefaultWriteOptions(), batch->GetWriteBatch());
+}
+
+rocksdb::Status BloomChain::bloomAdd(const Slice &bf_key, const std::string &item, int *ret) {
+  std::string bf_data;
+  rocksdb::Status s = storage_->Get(rocksdb::ReadOptions(), bf_key, &bf_data);
+  if (!s.ok()) return s;
+  BlockSplitBloomFilter block_split_bloom_filter;
+  block_split_bloom_filter.Init(std::move(bf_data));
+
+  uint64_t h = BlockSplitBloomFilter::Hash(item.data(), item.size());
+  bool found = block_split_bloom_filter.FindHash(h);
+
+  *ret = 0;
+  if (!found) {
+    *ret = 1;
+    block_split_bloom_filter.InsertHash(h);
+    auto batch = storage_->GetWriteBatchBase();
+    batch->Put(bf_key, block_split_bloom_filter.GetData());
+    return storage_->Write(storage_->DefaultWriteOptions(), batch->GetWriteBatch());
+  }
+
+  return rocksdb::Status::OK();
+}
+
+rocksdb::Status BloomChain::bloomCheck(const Slice &bf_key, const std::string &item, int *ret) {
+  LatestSnapShot ss(storage_);
+  rocksdb::ReadOptions read_options;
+  read_options.snapshot = ss.GetSnapShot();
+  std::string bf_data;
+  rocksdb::Status s = storage_->Get(read_options, bf_key, &bf_data);
+  if (!s.ok()) return s;
+  BlockSplitBloomFilter block_split_bloom_filter;
+  block_split_bloom_filter.Init(std::move(bf_data));
+
+  uint64_t h = BlockSplitBloomFilter::Hash(item.data(), item.size());
+  bool found = block_split_bloom_filter.FindHash(h);
+  *ret = found ? 1 : 0;
+
+  return rocksdb::Status::OK();
+}
+
+rocksdb::Status BloomChain::Reserve(const Slice &user_key, uint32_t capacity, double error_rate, uint16_t expansion) {
+  std::string ns_key = AppendNamespacePrefix(user_key);
+
+  LockGuard guard(storage_->GetLockManager(), ns_key);
+  BloomChainMetadata bloom_chain_metadata;
+  rocksdb::Status s = getBloomChainMetadata(ns_key, &bloom_chain_metadata);
+  if (!s.ok() && !s.IsNotFound()) return s;
+  if (!s.IsNotFound()) {
+    return rocksdb::Status::InvalidArgument("the key already exists");
+  }
+
+  return createBloomChain(ns_key, error_rate, capacity, expansion, &bloom_chain_metadata);
+}
+
+rocksdb::Status BloomChain::Add(const Slice &user_key, const Slice &item, int *ret) {
+  std::string ns_key = AppendNamespacePrefix(user_key);
+  LockGuard guard(storage_->GetLockManager(), ns_key);
+
+  BloomChainMetadata metadata;
+  rocksdb::Status s = getBloomChainMetadata(ns_key, &metadata);
+
+  if (s.IsNotFound()) {
+    rocksdb::Status create_s =
+        createBloomChain(ns_key, kBFDefaultErrorRate, kBFDefaultInitCapacity, kBFDefaultExpansion, &metadata);
+    if (!create_s.ok()) return create_s;
+  } else if (!s.ok()) {
+    return s;
+  }
```

Comment:
```suggestion

  if (s.IsNotFound()) {
    s =
        createBloomChain(ns_key, kBFDefaultErrorRate, kBFDefaultInitCapacity, kBFDefaultExpansion, &metadata);
  }
  if (!s.ok()) {
    return s;
  }
```

Would this a bit more clear?